plugins/resteasy-http/src/main/java/org/radargun/service/RESTEasyHTTPService.java

```diff
@@ -0,0 +1,198 @@
+
+   @Property(doc = "Expected cache name. Requests for other caches will fail. Defaults to 'default'.")
+   protected String cacheName = "default";
+
+   @Property(doc = "Semicolon-separated list of server addresses.", converter = AddressListConverter.class)
+   protected List<InetSocketAddress> servers;
+
+   @Property(doc = "Timeout for socket. Default is 30 seconds.", converter = TimeConverter.class)
+   protected long socketTimeout = 30000;
+
+   @Property(doc = "Timeout for connection. Default is 30 seconds.", converter = TimeConverter.class)
+   protected long connectionTimeout = 30000;
+
+   @Property(doc = ". Default is Integer.MAX_VALUE.")
+   protected int maxConnections = Integer.MAX_VALUE;
+
+   @Property(doc = ". Default is Integer.MAX_VALUE.")
+   protected int maxConnectionsPerHost = Integer.MAX_VALUE;
+
+   // Used to load balance requests across servers
+   protected AtomicInteger nextIndex = new AtomicInteger(0);
+
+   @ProvidesTrait
+   public RESTEasyHTTPOperations createOperations() {
+      return new RESTEasyHTTPOperations(this);
+   }
+
+   @ProvidesTrait
+   public Lifecycle getLifecycle() {
+      return this;
+   }
+
+   @Override
+   public synchronized void start() {
+      if (httpClient != null) {
+         log.warn("Service already started");
+         return;
+      }
+
+      httpClient = new ResteasyClientBuilder()
+            .httpEngine(new URLConnectionEngine())
+            .establishConnectionTimeout(connectionTimeout, TimeUnit.MILLISECONDS)
+            .socketTimeout(socketTimeout, TimeUnit.MILLISECONDS)
+            .connectionPoolSize(maxConnections)
+            .maxPooledPerRoute(maxConnectionsPerHost)
+            .hostnameVerification(HostnameVerificationPolicy.ANY)
+            .build();
+
+      if (username != null) {
+         BasicAuthentication auth = new BasicAuthentication(username, password);
+         httpClient.register(auth);
+      }
+
+   }
+
+   public String getContentType() {
+      return contentType;
+   }
+
+   public String getServerLocation() {
+      return serverLocation;
+   }
+
+   public ResteasyClient getHttpClient() {
+      return httpClient;
+   }
+
+   public String getUsername() {
+      return username;
+   }
+
+   public String getPassword() {
+      return password;
+   }
+
+   @Override
+   public synchronized void stop() {
+      if (httpClient == null) {
+         log.warn("Service not started");
+         return;
+      }
+      httpClient.close();
+      httpClient = null;
+   }
+
+   @Override
+   public synchronized boolean isRunning() {
+      return httpClient != null;
+   }
+
+   /**
+    * Distributes HTTP requests among servers.
+    * 
+    * @return InetSocketAddress of the next server to use.
+    */
+   public InetSocketAddress nextServer() {
+      return servers.get((nextIndex.getAndIncrement() & Integer.MAX_VALUE) % servers.size());
+   }
+
+   private static class AddressListConverter implements Converter<List<InetSocketAddress>> {
```

Comment:
Actually, this is more difficult. The code is shared with the SpyMemcachedService, and the issue is that the value of DEFAULT_PORT is different for memcached and REST clients. Converters are specified by class, so it isn't possible to parametrize this, unless you want to have 3 utility classes? (BaseAddressConverter, RESTAddressConverter, and MemcachedAddressConverter)